### ISSUE TITLE
x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by DanoThom** [istio/istio#37070]
### Bug Description

We've been running Istio 1.11x for a while now with no real issues at all, and then all of a sudden we started to receive the following issue when trying to deploy certain resources - in this case we noticed the issue when trying to deploy a virtual service and a destinationrule (note it first successfully applies the deployment and service without error):

`service/aegis-people unchanged
deployment.apps/aegis-people-1-0 configured
Error from server (InternalError): error when creating ".": Internal error occurred: failed calling webhook "validation.istio.io": Post "https://istiod.istio-system.svc:443/validate?timeout=10s": x509: certificate signed by unknown authority
Error from server (InternalError): error when creating ".": Internal error occurred: failed calling webhook "validation.istio.io": Post "https://istiod.istio-system.svc:443/validate?timeout=10s": x509: certificate signed by unknown authority
`

I'm able to deploy other similarly structured workloads without issues, the ingress gateway is still working and I'm able to make external calls into the cluster without issue either. 
As the error in question refers to the validate endpoint, initially in order to troubleshoot the issue I worked through this [doc](https://istio.io/latest/docs/ops/common-problems/injection/#x509-certificate-related-errors) and my md5s of the relevant certs all matched up correctly:

`danothom@DESKTOP-4RQPCGO:~$ kubectl -n istio-system get configmap istio-ca-root-cert -o jsonpath='{.data.root-cert\.pem}' | base64 -w 0 | md5sum
e5b164a7a7a79f0480f07a82a433e96c  -
danothom@DESKTOP-4RQPCGO:~$ kubectl get mutatingwebhookconfiguration istio-sidecar-injector -o yaml -o jsonpath='{.webhooks[0].clientConfig.caBundle}' | md5sum
e5b164a7a7a79f0480f07a82a433e96c  -
danothom@DESKTOP-4RQPCGO:~$ kubectl get validatingwebhookconfiguration istio-validator-istio-system -o yaml -o jsonpath='{.webhooks[0].clientConfig.caBundle}' | md5sum
e5b164a7a7a79f0480f07a82a433e96c  -
`

I then tried to cycle both the ingress gateway and the istiod pod in order but that didn't fix anything.
I then tried a few other things including trying to overwrite some of the Istio resources with the istio operator install but received the same validation x509 errors there too:
![image](https://user-images.githubusercontent.com/4973075/151592305-6f682885-69bf-4878-b69a-34404420ac07.png)

At this stage it appears something is broken at a fundamental level in my Istio install.

We're not sure what caused this issue all of a sudden as we haven't touched anything infrastructure related for a while. I've worked through a lot of the tangentially related issues in this Istio repo but am unable to find anything specifically relating to this exact scenario. My only options at this stage are to first try to rip out and then replace the root CA cert and cycle every thing in all namespaces, or finally just nuke everything and start again (both of which just seem way too extreme). I'd like to know that we could be confident that we'd be able to solve something like this in a production environment etc.

 Any ideas about this error and/or pointers in how to further troubleshoot this issue?

### Version

```prose
$ istioctl version
client version: 1.11.1
control plane version: 1.11.1
data plane version: 1.11.1 (55 proxies)
$ kubectl version --short
Client Version: v1.22.4
Server Version: v1.21.1
```

### Additional Information

[bug-report.log](https://github.com/istio/istio/files/7960451/bug-report.log)

**Comment by bianpengyuan** [comment]
Any suspicious logs from istiod?

**Comment by bianpengyuan** [comment]
The other possible issue is that your API server request go through some http proxy, like this similar issue: https://github.com/istio/istio/issues/36687

**Comment by hzxuzhonghu** [comment]
https://github.com/istio/istio/issues/36723

**Comment by bianpengyuan** [comment]
> #36723

I don't think there is any root update action? Since the root certs are all matched. Also the cert error here is from validating webhook, not proxy connecting to istiod, so I don't think #36723 is relevent?

**Comment by DanoThom** [comment]
Hi @bianpengyuan , thanks for taking a look.

I worked through the issues you referenced and I agree and don't think they are applicable due to the certs matching, some of the cluster still working and not being behind a proxy/having changed anything infrastructure related.

I've have however noticed a few other issues with Istio running in my cluster (AKS k8 1.21.1). When I try and change the debug log level for istiod I now get the following:
`istioctl proxy-config log "$(k get po -l app=istiod -n istio-system -o name).istio-system" --level debug
2022-01-31T08:55:14.913516Z     error   klog    an error occurred forwarding 61337 -> 15000: error forwarding port 15000 to pod 99fce8fdfd53157d8c2d36d14f3ec76913fae25d3f585402f6c22612227d5b6b, uid : failed to execute portforward in network namespace "/var/run/netns/cni-e2aa6508-b38d-c469-52c4-a4c0f44527b3": failed to dial 15000: dial tcp4 127.0.0.1:15000: connect: connection refused
Error: failed to execute command on Envoy: failure running port forward process: Post "http://127.0.0.1:61337/logging": EOF`

Also, there are defintely som interesting failures and errors around the mutating and validatingwebhookconfigurations in my istiod logs (which I've attached istiod after recently cycling the istiod pod

[istiod_dump.log](https://github.com/istio/istio/files/7969449/istiod_dump.log)

**Comment by bianpengyuan** [comment]
`istioctl proxy-config log` is meant for proxy log level tuning, not istiod log level.

The initial failure log seems fine, which is expected if Istio is newly installed. Do you performed some uninstall then resinstall operation?

**Comment by DanoThom** [comment]
Ok yes, that makes sense.

Istio hasn't been uninstalled in this cluster, it has however been updated and upgraded before a couple times using the Istio operator.

During my testing I was killing off a few of the webhooks and they were regenerated shortly afterwards. I then eventually deleted the istiod-istio-system validatingwebhookconfiguration resource (which was causing the issues) in my cluster and everything started working again. I then tried to recreate it by re-installing Istio over the the top - to no avail. So after trawling through the Go code for a bit I'm still unable to find out how to get the cluster to regenerate this webhook without a complete uninstall -> reinstall. 

As it appears to be performing quite a vital function I don't want to leave my cluster misconfigured or at risk of non-validated manifests etc.

Before I kill it all and re-install, I will make sure I fully scrub all traces of Istio and the webhooks or anything else that remains this time just to be sure.
